File: assemblyadapter-core/src/main/java/com/github/panpf/assemblyadapter/OnClickListener.java
package com.github.panpf.assemblyadapter;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface OnClickListener<DATA> {
    void onViewClick(@NonNull Context context, @NonNull View view, int position, @Nullable DATA data);
}

File: assemblyadapter-common-item/src/main/java/com/github/panpf/assemblyadapter/OnClickListener.kt
package com.github.panpf.assemblyadapter

import android.content.Context
import android.view.View

fun interface OnClickListener<DATA> {
    fun onClick(
        context: Context,
        view: View,
        bindingAdapterPosition: Int,
        absoluteAdapterPosition: Int,
        data: DATA
    )
}